Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ValueListGroup/ValueListForm.tsx

```diff
@@ -43,14 +43,17 @@ export default function ValueListForm(props: IProps) {
             e.push('Value must be less than 200 characters.');
         if (props.Record.AltValue != null && props.Record.AltValue.length > 200)
             e.push('Label must be less than 200 characters.');
-        
-    }, [props.Record])
+
+        props.SetErrors(e);
+    }, [props.Record, props.SetErrors])
```

Comment:
```suggestion
    }, [props.Record])
```
React is using a reference comparison for these. so if the parent component looks like `SetErrors={() => {...}}` that generates a new function on every render which would then run this Effect. It doesn't cause any visible issues but it is a performance drag. You don't need to re-run this every time `SetErrrors` changes.

Reply:
Okay I understand